funnel/models/event.py

```diff
@@ -0,0 +1,250 @@
+# -*- coding: utf-8 -*-
+import os
+import base64
+import logging
+from datetime import datetime
+from sqlalchemy.ext.associationproxy import association_proxy
+from . import db, BaseMixin
+from .space import ProposalSpace
+from .user import User
+
+__all__ = ['Event', 'TicketType', 'Participant', 'Attendee', 'SyncTicket', 'TicketClient']
+
+PRINTABLE_ASCII = map(chr, range(32, 127))
+
+
+def make_key():
+    return base64.urlsafe_b64encode(os.urandom(128))
+
+
+def make_public_key():
+    return make_key()[:8]
+
+
+def make_private_key():
+    return make_key()[:8]
+
+
+event_ticket_type = db.Table('event_ticket_type', db.Model.metadata,
+    db.Column('event_id', db.Integer, db.ForeignKey('event.id'), primary_key=True),
+    db.Column('ticket_type_id', db.Integer, db.ForeignKey('ticket_type.id'), primary_key=True),
+    db.Column('created_at', db.DateTime, default=datetime.utcnow, nullable=False)
+)
+
+
+class Event(BaseMixin, db.Model):
+    """
+    A discrete event under a proposal space.
+    For instance, a space could be associated with a workshop and a two-day conference.
+    The workshop constitutes as one event and each day of the conference
+    constitutes as an independent event.
+    This is done to allow distinguishing participants based on
+    on the tickets they have, given a participant may have a ticket
+    for only the workshop or a single day of the conference.
+    An event is associated with multiple ticket types,
+    which helps make the distinction between participants.
+    """
+    __tablename__ = 'event'
+
+    name = db.Column(db.Unicode(80), nullable=False)
```

Comment:
No, it isn't tied to the naming on Explara. Actually, nothing in this file is specific to Explara.